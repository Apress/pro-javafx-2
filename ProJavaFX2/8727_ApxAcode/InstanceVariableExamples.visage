/*
 * @test
 * @run
 */

class A {
  var i: Integer = 1024 on replace {
    println("Trigger in class definition: i = {i}.");
  }
}

var a1 = A {};
var a2 = A { i: 2048 };
var a3 = A {
  override var i = 3072 on replace {
    println("Trigger in object literal override: i = {i}.");
  }
};
var a4 = A {
  override var i = 4096 on replace {
    println("Trigger in object literal override: i = {i}.");
  }
  i: 5120
};

println("a1.i = {a1.i}");
println("a2.i = {a2.i}");
println("a3.i = {a3.i}");
println("a4.i = {a4.i}");
