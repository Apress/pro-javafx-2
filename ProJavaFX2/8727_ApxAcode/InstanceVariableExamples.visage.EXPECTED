Trigger in class definition: i = 1024.
Trigger in class definition: i = 2048.
Trigger in class definition: i = 3072.
Trigger in object literal override: i = 3072.
Trigger in class definition: i = 5120.
Trigger in object literal override: i = 5120.
a1.i = 1024
a2.i = 2048
a3.i = 3072
a4.i = 5120
